tjba/parser_test.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/dadosjusbr/coletores"
+	"github.com/stretchr/testify/assert"
+)
+
+const tjbaPayload = `[{
+	"matricula": 5014085,
+	"dataReferencia": 1577847600000,
+	"nome": "ADAILTON",
+	"lotacao": "COORDENACAO DE TRANSPORTE - SALVADOR",
+	"cargo": "MOTORISTA JUDICIÁRIO",
+	"status": "A",
+	"tipoServidor": "S",
+	"valorParadigma": 4896.59,
+	"valorVantagemPessoal": 9209.82,
+	"valorComissao": 0,
+	"valorIndenizacao": 1620,
+	"valorVantagemEventual": 0,
+	"totalCredito": 15726.41,
+	"valorPrevidencia": 1961.05,
+	"valorIR": 1856.3,
+	"valorDescontoDiverso": 195.86,
+	"valorRetencaoTeto": 0,
+	"totalDebito": 4013.21,
+	"valorLiquido": 11713.2,
+	"valorRemuneracaoOrigem": 0,
+	"valorDiaria": 0,
+	"valorGratificacao": 0,
+	"vantagensPessoais": [],
+	"vantagensEventuais": [],
+	"ano": null,
+	"mes": null,
+	"id": 5014085
+}]`
+
+func TestCreateTjbaEmployeesFromJSON(t *testing.T) {
+	employees, err := newTjbaEmployees(tjbaPayload)
+	assert.NoError(t, err)
+
+	var expectedTjbaEmployee = tjbaEmployee{
+		Reg:                 5014085,
+		Name:                "ADAILTON",
+		Workplace:           "COORDENACAO DE TRANSPORTE - SALVADOR",
+		Role:                "MOTORISTA JUDICIÁRIO",
+		Active:              "A",
+		Type:                "S",
+		Wage:                4896.59,
+		PersonalBenefits:    9209.82,
+		PositionOfTrust:     0,
+		PerksTotal:          1620,
+		EventualBenefits:    0,
+		IncomeTotal:         15726.41,
+		PrevContribution:    1961.05,
+		IncomeTax:           1856.3,
+		OtherDiscountsTotal: 195.86,
+		CeilRetention:       0,
+		DiscountTotal:       4013.21,
+		FundsTotal:          11713.2,
+		OriginPosition:      0,
+		Daily:               0,
+		Gratification:       0,
+	}
+	assert.Equal(t, employees[0], expectedTjbaEmployee)
+}
+
+func TestConvertFromTjbaEmployeeTypeToEmployeeType(t *testing.T) {
+	tests := []struct {
+		name     string
+		value    string
+		expected string
+	}{
+		{"servidor", "S", "servidor"},
+		{"membros", "D", "membro"},
+		{"pensionistas", "J", "membro"},
+		{"undefined", "W", ""},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := employeeType(tt.value); got != tt.expected {
+				t.Errorf("employeeType() = %v, want %v", got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestConvertFromTjbaEmployeeActiveToEmployeeActive(t *testing.T) {
+	tests := []struct {
+		name string
+		cat  string
+		want bool
+	}{
+		{"active", "A", true},
+		{"inactive", "I", false},
+		{"undefined", "?", false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := active(tt.cat); got != tt.want {
+				t.Errorf("active() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestFromTjbaEmployeeToEmployee(t *testing.T) {
+	tjbaEmployees, err := newTjbaEmployees(tjbaPayload)
+	assert.NoError(t, err)
+
+	employees := FromTjbaEmployeeToEmployee(tjbaEmployees)
+
+	var zero = 0.0
+	var expectedPersonalBenefits = 9209.82
```

Comment:
Como  esses números são usados apenas aqui, por favor coloca eles direto no expectedEmployee. Já vai ficar claro que  tudo que está láé expected.

Reply:
Foi o que tentei fazer mas peguei um `cannot use 0 (type float64) as type *float64 in field value`. Também, pensei que era a prática certa para resolver isso (vi [aqui](https://github.com/dadosjusbr/coletores/blob/f94cbbc5b0d5684eff79addbd52b58d7316ba3d2/trepb/parser_test.go#L51) também). Qual a forma correta de resolver?

Reply:
Eita pega, esqueci que são ponteiros. Foi mal a sugestão foi errada!